### PR TITLE
CACHE YA LATER! (caching the heatmap computation every week) 

### PR DESCRIPTION
**Pull request by dfarooq610**

### CODE REVIEW

File: packages/server/src/course/heatmap.service.ts [hunk 1]
```diff
@@ -228,6 +242,6 @@ export class HeatmapService {
     autoExit: true,
   })
   async create(): Promise<void> {
-    const h = await this.getHeatmapFor(4);
+    const h = await this._getHeatmapFor(4);
```
Comment (willstenzel) [inline packages/server/src/course/heatmap.service.ts:245]: This wasn't from this PR but why is `4` is hardcoded in the `heatmap:generate` command?

**Review by willstenzel — APPROVED** [review]
Nice work! 

I'd be curious as to what happens to the performance of the endpoint once this goes to prod.

File: packages/server/src/course/heatmap.service.ts (hunk 1 above)
Comment (NEUDitao) [reply packages/server/src/course/heatmap.service.ts:245]: ^^^^ it seems to be a dev thing but... still concerning

**Review by NEUDitao — APPROVED** [review]

File: packages/server/src/course/heatmap.service.ts (hunk 1 above)
Comment (dfarooq610) [reply packages/server/src/course/heatmap.service.ts:245]: I think this was a command thing, in a separate PR I'm gonna update this command

**Pull request merged**